src/Features/CSharp/Completion/CompletionProviders/ObjectCreationCompletionProvider.ItemRules.cs

```diff
@@ -15,10 +16,22 @@ internal class ItemRules : AbstractSymbolCompletionItemRules
 
             public override bool? IsCommitCharacter(CompletionItem completionItem, char ch, string textTypedSoFar)
             {
-                // TODO(cyrusn): We could just allow the standard list of completion characters.
-                // However, i'd like to see what the experience is like really filtering down to the set
-                // of things that is allowable.
-                return ch == ' ' || ch == '(' || ch == '{' || ch == '[';
+                if (ch == '{')
+                {
+                    // SPECIAL: If the preselected symbol is System.Object, don't commit on '{'.
+                    // Otherwise, it is cumbersome to type an anonymous object when the target type is object.
+                    // The user would get 'new object {' rather than 'new {'. Since object doesn't have any
+                    // properties, the user never really wants to commit 'new object {' anyway.
+                    var namedTypeSymbol = (completionItem as SymbolCompletionItem)?.Symbols.FirstOrDefault() as INamedTypeSymbol;
+                    if (namedTypeSymbol?.SpecialType == SpecialType.System_Object)
+                    {
```

Comment:
Because this rule is only for items returned by the ```ObjectCreationCompletionProvider```, all of the items are preselected. Other "objects" that appear in the list will come from different providers (like the symbol or keyword completion providers) and not have this rule. Thus, they continue to commit on '{'.